packages/react/src/ReactChildren.js

```diff
@@ -229,11 +229,21 @@ function mapIntoArray(
             childKey,
         );
         if (__DEV__) {
-          if (nameSoFar !== '' && mappedChild.key == null) {
-            // We need to validate that this child should have had a key before assigning it one.
-            if (!newChild._store.validated) {
-              // We mark this child as having failed validation but we let the actual renderer
-              // print the warning later.
+          // If `child` was an element without a `key`, we need to validate if
+          // it should have had a `key`, before assigning one to `mappedChild`.
+          // $FlowFixMe[incompatible-type] Flow incorrectly thinks React.Portal doesn't have a key
```

Comment:
It is weird that React.Children doesn't operate on portals. They don't get assigned keys.

Reply:
To be honest, I copied this from the `$FlowFixMe` comments in the lines above. 😅

Reply:
I know but I found it interesting that 1) we didn't refine the type above 2) that we're checking isValidElement at all.